src/test/java/uk/gov/justice/digital/config/BaseSparkTest.java

```diff
@@ -0,0 +1,193 @@
+package uk.gov.justice.digital.config;
+
+
+
+import static org.apache.spark.sql.functions.col;
+import static org.apache.spark.sql.functions.from_json;
+import static org.apache.spark.sql.functions.lit;
+import static org.apache.spark.sql.functions.schema_of_json;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.types.DataType;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.io.TempDir;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.profile.ProfileCredentialsProvider;
+
+public abstract class BaseSparkTest {
+
+	@TempDir
+	Path folder;
+	
+	protected SparkSession spark;
+	
+
+	protected String accessKey;
+	protected String secretKey;
+	
+
+	@BeforeEach
+	public void before() {
+		
+		spark = SparkSession.builder()
+			    .appName("test")
+			    .enableHiveSupport()
+				.config("spark.master", "local")
+				// important delta configurations
+				// =================================
+				// these need to be in the cloud-platform
+				.config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
+			    .config("spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog")
+			    .config("spark.databricks.delta.schema.autoMerge.enabled", true)
+			    .config("spark.databricks.delta.optimizeWrite.enabled", true)
+			    .config("spark.databricks.delta.autoCompact.enabled", true)
+			    // ============================
+			    // these are needed for test but NOT for live
+			    // the manifest needs a HiveContext and this handles a separate one for each test
+			    // otherwise we do an inmem one : jdbc:derby://localhost:1527/memory:myInMemDB;create=true
+			    .config("javax.jdo.option.ConnectionURL", "jdbc:derby:;databaseName=" +
+						folder.toFile().getAbsolutePath() + "/metastore_db_test;create=true")
+			    .getOrCreate();
+
+		try {
+			final AWSCredentials credentials = new ProfileCredentialsProvider("moj").getCredentials();
```

Comment:
If we're just running spark for local tests we don't need to have any aws code afaik. remove this since I don't think it's necessary (unless I'm missing something?).